pyquil/paulis.py

```diff
@@ -715,7 +718,7 @@ def from_compact_str(cls, str_pauli_sum):
         return cls(terms).simplify()
 
 
-def simplify_pauli_sum(pauli_sum):
+def simplify_pauli_sum(pauli_sum: PauliSum) -> PauliSum:
```

Comment:
Why are some of these `PauliSums` in quotes, while others are not?

Reply:
Oh, it is because `simplify_pauli_sum` is defined outside of the class definition of `PauliSum`, at which point it is fine to just say `PauliSum` without quotation.

Reply:
The ones with quotation are the methods of `PauliSum` itself.

Reply:
Oh that's weird. A type can't refer to ... itself?

Reply:
Not in current mypy, though there is [PEP 563](https://www.python.org/dev/peps/pep-0563/) available in python 3.7 via `from __future__ import annotations`, and available in python 4.0 by default.